### ISSUE TITLE
jp-Osaka city 

### ISSUE THREAD

**Issue by nyampire** [openaddresses/openaddresses#877]
Osaka City, Japan has published their Address data under CC-BY2.1 JP.
I guess we could include it in this repository.

## Data
* Address dataset
* http://www.city.osaka.lg.jp/contents/wdu090/opendata/mapnavoskdat_csv/jusho.csv

EPSG is 2448.
UTF-8 encoded.

### 区コード (suburb code): 

last 2 digit of CODE_ME in this dataset.
http://www.city.osaka.lg.jp/contents/wdu090/opendata/mapnavoskdat_csv/kucode.csv

e.g. 
CODE_PARENT,CODE_ME,NAME_ME,
27,27001,北区,

27001 -> 1 = 区コード column in Address dataset

### 町コード (neighbourhood code): 
last 4 digit of CODE_ME in this dataset.
http://www.city.osaka.lg.jp/contents/wdu090/opendata/mapnavoskdat_csv/chocode.csv

e.g.
CODE_PARENT,CODE_ME,NAME_ME
27005,270051496,備後町

270051496 -> 1496 = 町コード column in Address dataset

### Original Link page (& License)
http://www.city.osaka.lg.jp/toshikeikaku/page/0000250227.html

**Comment by geobrando** [comment]
Hey @nyampire.  Do you know how this differs from the [current Osaka source](https://github.com/openaddresses/openaddresses/blob/984589b4b3002c92b8e9e56ca73c611c9f1f778b/sources/jp/osaka.json)?

**Comment by nyampire** [comment]
My understanding.
current Osaka source is from Ichi Sansho Joho.
Those indicate the delegation points of "block number" (similar to street name) polygon.
So not include the data of house number points.

This data set include real dump of "house number".

**Comment by geobrando** [comment]
OK. So is the current JSON that has `number` and `street` fields incorrect?

**Comment by geobrando** [comment]
@nyampire Looks like these were added by @sbma44 [here](https://github.com/openaddresses/openaddresses/pull/707)

**Comment by sbma44** [comment]
yeah, the old source is a nationwide block-level source from MLIT. address-level data is a very desirable supplement, though you will need both to build a decent japanese geocoder.

**Comment by justinelliotmeyers** [comment]
@sbma44 the Japan data we have now isn't correct, to the best of my knowledge. Last time I looked it was for blocks, hence not actual addresses. @nyampire can we revisit this? I just downloaded the Jusho file - I can try to wrap my head around where to join data from if you can help? I tried to get the ball rolling on this stuff here: https://github.com/openaddresses/openaddresses/issues/3628  Lets do this!

**Comment by nyampire** [comment]
@justinelliotmeyers I see. I'll wrap up current situation roughly. Please wait a while.

More general information about Japanese situation is [aggregated here](https://qiita.com/nyampire/items/d74dcde6e57f793ab0c1) (it is 3 years ago, and written in Japanese though).

Just a confirmation, we are collecting "addresses points" not "Gaiku (block-level data)" like Ichi-Sasho-Joho.

**Comment by justinelliotmeyers** [comment]
@nyampire yes, addresses/ address points. Currently for Japan, we support a level above what we ideally want for this project. The data I believe you have submitted is what we need. 

Wow, your blog post is amazing! I will have to review in detail this weekend. But yes, we want individual house addresses, and the most detailed data possible. Lets stay connected and try to create some data!

**Comment by nyampire** [comment]
@justinelliotmeyers OK, I understood.

And sorry, this is not a direct issue about this thread.
Does this repo (OpenAddresses) have any aggregated documents to explain each country's address scheme and/or structure? (similar to ISO/TC211's documents)

In OSM wiki, I've written an article to explain the basic structure of [the Japanese "block-based" address system](https://wiki.openstreetmap.org/wiki/JA:%E4%BD%8F%E6%89%80#.E8.A1.97.E5.8C.BA_Block_number_address_system.E3.81.AB.E3.81.A4.E3.81.84.E3.81.A6). I think some notes like it could be worth to get an insight of every country's situation.

**Comment by justinelliotmeyers** [comment]
@nyampire Downloaded data and put together. Compare below. Currently we have block level points (align inside 街区ポリゴン). The new data you have submitted is a lower level (住所データ). I need to examine the attributes to see if these are addresses we would expect to see. 
![jp_old](https://user-images.githubusercontent.com/5324316/77824530-cdbab180-70d9-11ea-9985-dd1f3e054661.png)

![jp_new](https://user-images.githubusercontent.com/5324316/77824537-d317fc00-70d9-11ea-9a72-ecc148a8924d.png)

**Comment by justinelliotmeyers** [comment]
an example, do these have building addresses? I'm not sure if this is the official address where people would send mail to, or drive to, but something like: https://www.city.osaka.lg.jp/fukushutosuishin/page/0000487546.html

〒530-8201　大阪市北区中之島1丁目3番20号

**Comment by nyampire** [comment]
Thank you for examining. And sorry, I'm not sure if I could understand your mention correctly...

The new dataset is the number for the buildings. 号 (号・番号 column in this dataset) is the number for the buildings.

For example, the address for townhall you mentioned above is separatable as following.

〒530-8201 / 大阪市 / 北区 / 中之島 / 1丁目 / 3番 / 20号

〒530-8201 = zip/postal code
大阪市 = city
北区 = suburb (or called as ward) name
中之島 = neighbourhood (or district) name
1丁目 = neighbourhood number
3番 = block number
20号 = house number

It is very confusing the neighbourhood name and neighbourhood number is sometimes combined as name (e.g. officially 中之島1丁目 is the name in municipality's documents. But some data like this or [IMI](https://imi.go.jp/goi/imi-about-en/) (federal standards, like NIEM in US) is separating it as name and number)

**Comment by nyampire** [comment]
As my understanding, following point data is a center point of the block.  (3番 = block number above)
This is not a house number point data.

![image](https://user-images.githubusercontent.com/428218/77879920-7150c100-7296-11ea-9104-1b4b97c91694.png)

**Comment by justinelliotmeyers** [comment]
![osaka_review](https://user-images.githubusercontent.com/5324316/77915151-f4f1ba00-7264-11ea-8bf6-a6229e6ae4a5.png)

Uploading a csv in raw EPSG is 2448 and in 4326 (WGS84). Could you let me know the structure you believe we should implement?


[osaka_2448.zip](https://github.com/openaddresses/openaddresses/files/4402917/osaka_2448.zip)
[osaka_4326.zip](https://github.com/openaddresses/openaddresses/files/4402919/osaka_4326.zip)

**Comment by nyampire** [comment]
@justinelliotmeyers I could load 4326 in QGIS, but couldn't for 2448. Which is the correct X,Y column for that (X1, X2, GX)? And is EPSG 2448 correct for this csv?

Anyway, 4326 looks good for me at least.

**Comment by nyampire** [comment]
in the context of data structure, how about this attribute table?

|address structure | column name |
|----|----|
|X | POINT_X|
|Y | POINT_Y|
|subsurb name |NAME_ME|
|subsurb name in KANA | KANA_ME|
|neighbourhood name | NAME_ME_1|
|neighbourhood name in KANA | KANA_ME_1|
|block number | 街区コード|
|house number | 号・番号|

**Comment by justinelliotmeyers** [comment]
@sbma44 thoughts about starting with this as a new source for revamping what we have for Japan? If you aren't a good person to ping let me know, and I can reach out to Ingalls and see.  Thanks!

**Comment by sbma44** [comment]
I’m too far removed to have an informed opinion, @justinelliotmeyers. But I’ve pinged some of our Japanese address experts in the hopes they might be able to weigh in (@ingalls is worth bothering too)

**Comment by justinelliotmeyers** [comment]
thanks @sbma44 just like to keep people involved when/ where I can. I'll ping Ingalls.

**Comment by justinelliotmeyers** [comment]
@ingalls 
see structure from above:

![osaka](https://user-images.githubusercontent.com/5324316/79818208-3f8ebf80-8355-11ea-9d67-fb3c0f60ae04.png)
